### ISSUE TITLE
Sync API Entries Not of Type Entry

### ISSUE THREAD

**Issue by hugobast** [contentful/contentful.js#280]
## Expected Behavior

```javascript
client.sync({ initial: true })
  .then(response => {
    response.entries.forEach(entry => {
      const plain = entry.toPlainObject() // <- This should be an Entry?
```
I am using the 7.1.0-beta0 SDK according to your docs:

- `response` ([link](https://contentful.github.io/contentful.js/contentful/7.1.0-beta0/ContentfulClientAPI.html#.sync)) should a `SyncCollection`
- `response.entries` ([link](https://contentful.github.io/contentful.js/contentful/7.1.0-beta0/Sync.html#.SyncCollection) should be an `Array.<Entities.Entry>`
- `entry` ([link](https://contentful.github.io/contentful.js/contentful/7.1.0-beta0/Entities.html#.Entry)) should now be an `Entry`

## Actual Behavior

The `entry` variable doesn't have a `toPlainObject` on it, and I'm not able to use this object to store it anywhere since it contains circular references.

## Possible Solution

I've already tried to import `toPlainObject` from contentful-sdk-core and pass the data in though this adds the `toPlainObject` function on the object passed into it, calling it still maintains the circular references.

## Steps to Reproduce

I think simply using the client's sync API call, grabbing entries from the response and the first entry more specifically should be enough to reproduce that the object doesn't have a `toPlainObject` on it.

## Environment

-   **Language Version**:  v8.9.1
-   **Package Manager Version**: yarn 1.9.4
-   **Operating System**:  Darwin machine.local 16.7.0 Darwin Kernel Version 16.7.0
-   **Package Version**:  7.1.0-beta0 and 7.0.3
-   **Which API are you using?**:  Delivery

**Comment by hugobast** [comment]
@phoebeschmidt amazing thank you so much!

**Comment by phoebeschmidt** [comment]
Not sure why Semantic Release didn't add a comment here BUT:
This fix is released as part of v7.0.4 🎉
